Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -1013,6 +1028,7 @@ private void analysis_setSubscriptions() {
 
       final Map<String, List<String>> subscriptions = new THashMap<String, List<String>>();
       subscriptions.put(AnalysisService.HIGHLIGHTS, myVisibleFiles);
+      subscriptions.put(AnalysisService.IMPLEMENTED, myVisibleFiles);
```

Comment:
Got it. It is not jar that breaks work with SDK 1.12. It is this line. After receiving unknown subscription key DAS from 1.12 stops sending any file-level notifications. At least on Windows. On your screenshot that might be cached results received before changing SDK, or just different behavior on Mac.
I'll surround this line with if.